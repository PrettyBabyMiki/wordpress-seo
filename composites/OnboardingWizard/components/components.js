import Choice from "./Choice";
import Input from "./Input";
import HTML from "./Html";
import MailchimpSignup from "./custom_components/MailchimpSignup";
import PublishingEntity from "./custom_components/PublishingEntity";
import PostTypeVisibility from "./custom_components/PostTypeVisibility";
import ConnectGoogleSearchConsole from "./custom_components/ConnectGoogleSearchConsole";

let Components = {
	Choice,
	Input,
	HTML,
	MailchimpSignup,
	PostTypeVisibility,
	ConnectGoogleSearchConsole,
	PublishingEntity,
};

export default Components;
