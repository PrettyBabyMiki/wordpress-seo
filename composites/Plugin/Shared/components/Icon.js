import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import _omit from "lodash/omit";

/**
 * Returns the Icon component.
 *
 * @param {object} props Component props.
 *
 * @returns {ReactElement} Icon component.
 */
export const Icon = ( props ) => {
	const IconComponent = styled( props.icon )`
		width: ${ props.size };
		height: ${ props.size };
		fill: ${ props.color };
		flex: none;
	`;

	// Remove the props that are no longer needed.
	const newProps = _omit( props, [ "icon", "size", "color" ] );

	return <IconComponent role="img" aria-hidden="true" focusable="false" { ...newProps } />;
};

Icon.propTypes = {
	icon: PropTypes.func.isRequired,
	color: PropTypes.string.isRequired,
	size: PropTypes.string,
};

Icon.defaultProps = {
	size: "16px",
};

/**
 * Returns the IconWithAriaLabel component.
 *
 * Safari 11 doesn't expose to browsers the ARIA role set on SVG elements.
 *
 * @param {object} props Component props.
 *
 * @returns {ReactElement} IconWithAriaLabel component.
 */
export const IconWithAriaLabel = ( props ) => {
	return <span role="img" aria-label={ props.ariaLabel }>
		<Icon
			icon={ props.icon }
			color={ props.color }
			size={ props.size }
			role="presentation"
		/>
	</span>;
};

IconWithAriaLabel.propTypes = {
	icon: PropTypes.func.isRequired,
	color: PropTypes.string.isRequired,
	size: PropTypes.string.isRequired,
	ariaLabel: PropTypes.string.isRequired,
};
