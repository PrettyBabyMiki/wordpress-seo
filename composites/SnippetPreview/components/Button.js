import styled from "styled-components";

/**
 * @summary:
 * Returns a basic styled button
 *
 * @param {object} props Component props
 *
 * @returns {ReactElement} Styled button
 */
const Button = styled.button`
	display: inline-block;
	padding: 8px 10px;
    border: 1px solid #dbdbdb;
    border-radius: 4px;
    color: #555;
    background: #f7f7f7;
    font-size: 0.8rem;
    line-height: 15px;
    cursor: pointer;
    box-sizing: border-box;
    font-family: inherit;
    font-weight: inherit;
    
    :focus {
		border-color: #5b9dd9;
		outline: none;
		color: #23282d;
		background-color: #fafafa;
		box-shadow: 0 0 3px rgba(0, 115, 170, 0.8);
    }
`;

export default Button;
