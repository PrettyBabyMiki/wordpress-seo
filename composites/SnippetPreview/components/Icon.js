import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

/**
 * Returns the Icon component.
 *
 * @param {object} props Component props.
 *
 * @returns {ReactElement} Icon component.
 */
export const Icon = ( props ) => {
	const IconComponent = styled( props.icon )`
		float: left;
		width: ${ props => props.iconSize }px;
		height: ${ props => props.iconSize }px;
		fill: ${ props => props.iconColor };
	`;

	return <IconComponent aria-hidden="true" iconSize={ props.iconSize } iconColor={ props.iconColor } />;
};

Icon.propTypes = {
	icon: PropTypes.func.isRequired,
	iconColor: PropTypes.string.isRequired,
	iconSize: PropTypes.number,
};

Icon.defaultProps = {
	iconSize: 16,
};
