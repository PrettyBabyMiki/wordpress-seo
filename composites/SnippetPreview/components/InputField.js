import React from "react";
import { EditorState, Modifier, Editor } from "draft-js";
import colors from "../../../style-guide/colors.json";
import styled from "styled-components";

const styles = {
	root: {
		fontFamily: "Arial, Helvetica, sans-serif",
		fontSize: 14,
	},
	editor: {
		cursor: "text",
		height: 16,
		border: "1px solid",
		borderColor: colors.$color_grey,
		width: "20%",
		padding: "3px 5px",
		margin: "5px 1px 1px 1px",
		backgroundColor: colors.$color_white,
		overflow: "hidden",
	},
};

const EditorContainer = styled.div`
    .public-DraftStyleDefault-block {
        white-space: nowrap;
        word-wrap: normal;
    }
`;

export default class InputField extends React.Component {
	constructor( props ) {
		super( props );
		this.state = { editorState: EditorState.createEmpty() };
	}

	render() {
		return (
			<div style={ styles.root }>
				<div style={ styles.editor } onClick={ this.focus.bind( this ) }>
					<EditorContainer>
						<Editor
							editorState={ this.state.editorState }
							onChange={ this.onChange.bind( this ) }
							placeholder={ this.props.placeholder }
							ref="editor"
						/>
					</EditorContainer>
				</div>
			</div>
		);
	}

	onChange( editorState ) {
		this.setState( { editorState } );
	}

	focus() {
		this.refs.editor.focus();
	}
}
