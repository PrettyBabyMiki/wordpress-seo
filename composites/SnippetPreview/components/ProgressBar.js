import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import colors from "../../../style-guide/colors.json";

const ProgressBarContainer = styled.progress`
	box-sizing: border-box;
	width: 100%;
	height: 8px;
	display: block;
	margin-top: 5px;
	border: none;
	-webkit-appearance: none;
	
	::-webkit-progress-bar {
	   	background-color: ${colors.$color_progress_background};
	   	color: ${( { color } ) => color };
		border: 1px solid ${colors.$color_input_border};
	}
	
	::-webkit-progress-value {
		background-color: ${ props => props.progressColor };
	}
`;

/**
 * The ProgressBar component
 *
 * @summary
 * A component that displays a progress bar
 */
class ProgressBar extends React.Component {
	render() {
		const { max, value, progressColor } = this.props;
		return (
			<ProgressBarContainer progressColor={ progressColor } aria-hidden="true" max={max} value={value}>
			</ProgressBarContainer>
		);
	}
}

ProgressBar.defaultProps = {
	max: 1,
	value: 0,
	progressColor: colors.$color_good,
}

ProgressBar.propTypes = {
	max: PropTypes.number,
	value: PropTypes.number,
	progressColor: PropTypes.string,
}

/**
 * Returns the ProgressBar component.
 *
 * @returns {ReactElement} The ProgressBar component.
 */
export default ProgressBar;
