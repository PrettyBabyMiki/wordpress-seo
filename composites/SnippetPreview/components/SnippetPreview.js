import React from "react";
import styled from "styled-components";

import { PrimaryCell, IconCell, responsiveHeaders } from "../../basic/Table/Cell";
import { Row, RowMobileCollapse } from "../../basic/Table/Row";
import { ListTable, ZebrafiedListTable } from "../../basic/Table/ListTable";

export const SnippetPreviewDiv = styled.div`
	height: 700px;
	width: 100%;
	background-color: white;
`;



let PrimaryCellResponsive = responsiveHeaders( PrimaryCell );
let rowProps = [];

/**
 * Returns the SnippetPreview component.
 *
 * @returns {ReactElement} The SnippetPreview component.
 */
export default function SnippetPreview() {
	return(
		<SnippetPreviewDiv>
			<br /><br /><br />
			<ListTable>
				<RowMobileCollapse { ...rowProps }>
					<PrimaryCell headerLabel="Test column">
						This is a text column
					</PrimaryCell>
					<PrimaryCell headerLabel="Test column2">
						This is a text column
					</PrimaryCell>
					<PrimaryCell headerLabel="Test column3">
						This is a text column
					</PrimaryCell>
					<PrimaryCell headerLabel="Test column4">
						This is a text column
					</PrimaryCell>
					<PrimaryCellResponsive headerLabel="responsiveHeaders">
						This is a primary cell with responsive headers
					</PrimaryCellResponsive>
					<IconCell headerLabel="IconCell">
						Icon
					</IconCell>
				</RowMobileCollapse>
			</ListTable>
			<br /><br /><br /><br /><br /><br />
			<ZebrafiedListTable>
				<Row { ...rowProps }>
					<PrimaryCell ellipsis={ true } headerLabel="Test column">
						This is a text column
					</PrimaryCell>
					<PrimaryCell ellipsis={ true } headerLabel="Test column2">
						This is a text column
					</PrimaryCell>
					<PrimaryCell ellipsis={ true } headerLabel="Test column3">
						This is a text column
					</PrimaryCell>
					<PrimaryCell ellipsis={ true } headerLabel="Test column4">
						This is a text column
					</PrimaryCell>
					<PrimaryCellResponsive ellipsis={ true } headerLabel="responsiveHeaders">
						This is a primary cell with responsive headers
					</PrimaryCellResponsive>
					<IconCell ellipsis={ true } headerLabel="IconCell">
						Icon
					</IconCell>
				</Row>
				<Row { ...rowProps }>
					<PrimaryCell ellipsis={ true } headerLabel="Test column">
						This is a text column
					</PrimaryCell>
					<PrimaryCell ellipsis={ true } headerLabel="Test column2">
						This is a text column
					</PrimaryCell>
					<PrimaryCell ellipsis={ true } headerLabel="Test column3">
						This is a text column
					</PrimaryCell>
					<PrimaryCell ellipsis={ true } headerLabel="Test column4">
						This is a text column
					</PrimaryCell>
					<PrimaryCellResponsive ellipsis={ true } headerLabel="responsiveHeaders">
						This is a primary cell with responsive headers
					</PrimaryCellResponsive>
					<IconCell ellipsis={ true } headerLabel="IconCell">
						Icon
					</IconCell>
				</Row>
			</ZebrafiedListTable>
		</SnippetPreviewDiv>
	);
}
