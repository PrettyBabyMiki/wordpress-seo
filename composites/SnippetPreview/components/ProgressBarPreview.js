import React from "react";
import styled from "styled-components";
import ProgressBar from "./ProgressBar";

export const Preview = styled.div`
	height: 700px;
	width: 100%;
	background-color: white;
`;

class ProgressBarTest extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			value: 0,
		};
	}

	render() {
		console.log(this.state.value);
		return (
			<Preview>
				<button onClick={() => this.setState( { value: this.state.value - 10 } ) }>-10%</button>
				<button onClick={() => this.setState( { value: this.state.value + 10 } ) }>+10%</button>
				<ProgressBar max={100} value={this.state.value}/>
			</Preview>
		);
	}
}

export default ProgressBarTest;