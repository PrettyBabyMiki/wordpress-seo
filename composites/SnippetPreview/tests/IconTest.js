import React from "react";
import renderer from "react-test-renderer";

import Edit from "../../../style-guide/svg/edit.svg";
import { Icon } from "../components/Icon";

test( "the Icon matches the snapshot", () => {
	const component = renderer.create(
		<Icon icon={ Edit } color="black"  />
	);

	let tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );
