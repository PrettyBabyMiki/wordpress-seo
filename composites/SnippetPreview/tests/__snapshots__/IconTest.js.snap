// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Icon matches the snapshot 1`] = `
<svg
  className="sc-bdVaJa jEZkCV"
  color="black"
  icon={[Function]}
  size={16}
/>
`;
